### PR TITLE
fix(synthetics): default execution role breaks in non aws partitions

### LINKED ISSUE #12094: (synthetics): Default role breaks in non aws partitions
Synthetics Canary default execution role hard codes `'arn:aws:logs:::*'` in the IAM policy which does not work in non AWS partitions (i.e aws-cn or aws-us-gov)

### Reproduction Steps
Synthesize the following code (`cdk synth`) for **cn-north-1**
```
import * as synthetics from '@aws-cdk/aws-synthetics';

const canary = new synthetics.Canary(this, 'MyCanary', {
  schedule: synthetics.Schedule.rate(Duration.minutes(5)),
  test: Test.custom({
    code: synthetics.Code.fromAsset(path.join(__dirname, 'canary')),
    handler: 'index.handler',
  }),
  runtime: synthetics.Runtime.SYNTHETICS_NODEJS_2_0,
});
````

### What did you expect to happen?
The default execution role IAM policy should contain a partition aware log access policy
```
{
  "Action": [
    "logs:CreateLogStream",
    "logs:CreateLogGroup",
    "logs:PutLogEvents"
  ],
  "Effect": "Allow",
  "Resource": { "Fn::Join": ["", ["arn:", {"Ref": "AWS::Partition"}, ":logs:::*"]] }
}
```

### What actually happened?
The default execution role contains an IAM policy which has **aws** hardcoded and isn't partition aware
```
{
  "Action": [
    "logs:CreateLogStream",
    "logs:CreateLogGroup",
    "logs:PutLogEvents"
  ],
  "Effect": "Allow",
  "Resource": "arn:aws:logs:::*"
}
```

### Environment

  - **CDK CLI Version  :** 1.76.0
  - **Framework Version: Version:** monocdk 1.77.0
  - **Node.js Version:** 12.9.1
  - **OS               :**  macOS 10.15.7
  - **Language (Version):** JavaScript 2020

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by Khufu-I**
Canary default execution role should be partition aware instead of hardcoding aws.

Fixes #12094

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by Khufu-I** [comment]
Is there anything else that needs to be done to get this merged?

**Comment by Khufu-I** [comment]
@NetaNir could we get this fix merged?

File: packages/@aws-cdk/aws-synthetics/lib/canary.ts [hunk 1]
```diff
@@ -309,6 +309,7 @@ export class Canary extends cdk.Resource {
    * Returns a default role for the canary
    */
   private createDefaultRole(prefix?: string): iam.IRole {
+    const { partition } = cdk.Stack.of(this);
```
Comment (NetaNir) [inline packages/@aws-cdk/aws-synthetics/lib/canary.ts:312]: `stack.of` returns the encompassing stack, not the partition. This should be:

```
cdk.Stack.of(this).partition
``` 
Was this change successfully deployed?

**Review by NetaNir — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-synthetics/lib/canary.ts (hunk 1 above)
Comment (Khufu-I) [reply packages/@aws-cdk/aws-synthetics/lib/canary.ts:312]: This is using [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring) to only select the partition from the returned stack object similar to [this code](https://github.com/aws/aws-cdk/blob/b96efd8aaa143845b9fe315a9ee1e8398c4d83c2/packages/%40aws-cdk/aws-route53-targets/lib/bucket-website-target.ts#L14). I've run the tests successfully but I've not deployed this exact CR to AWS Cloudformation. Could we merge without doing a deployment?

Comment (NetaNir) [reply packages/@aws-cdk/aws-synthetics/lib/canary.ts:312]: How were you able to update the `integ.canary.expected.json` file without deploying it?

Comment (Khufu-I) [reply packages/@aws-cdk/aws-synthetics/lib/canary.ts:312]: I made the update based on similar changes I made to my Cloudformation project and deployed to AWS. I'm new to this Github repository so I'm not sure how to reference an unpublished version of AWS CDK from Github and deploy to my AWS account.

Comment (NetaNir) [reply packages/@aws-cdk/aws-synthetics/lib/canary.ts:312]: This file needs to be updated by executing the integration test - **not manually**. The stack defined `integ.canary.ts` will be deployed and if successful the expectation file will be updated. 

From the @aws-cdk/aws-synthetics folder executing:

```
yarn integ integ.canary.js
```

Will deploy the stack and afterward update the file.

Comment (Khufu-I) [reply packages/@aws-cdk/aws-synthetics/lib/canary.ts:333]: Thanks for walking me through the process. I successfully deployed the changes to my AWS account and the integration test passed succesfully

```
gitpod /workspace/aws-cdk/packages/@aws-cdk/aws-synthetics $ yarn integ integ.canary.js      
  yarn run v1.22.5            
    $ cdk-integ integ.canary.js                  
      Synthesizing integ.canary.js.                        
        Selected stack: canary-one                              
          Deploying integ.canary.js...                                    
            Deployment succeeded, updating snapshot.
Cleaning up.
Done in 129.24s.
```

There were no additional changes after the snapshot was updated.

**Review by NetaNir — APPROVED** [review]

**Pull request merged**